mypy/type_visitor.py

```diff
@@ -159,7 +159,18 @@ def visit_deleted_type(self, t: DeletedType) -> Type:
         return t
 
     def visit_instance(self, t: Instance) -> Type:
-        return Instance(t.type, self.translate_types(t.args), t.line, t.column)
+        final_value = None  # type: Optional[LiteralType]
+        if t.final_value is not None:
+            raw_final_value = t.final_value.accept(self)
+            assert isinstance(raw_final_value, LiteralType)
+            final_value = raw_final_value
```

Comment:
It's because `t.final_value.accept(self)` has a return type of `Type`, which means we wouldn't be able to assign it to `final_value` without raising an error or without casting.

I can replace this entire if statement with a cast if you want, similar to what we're doing in `visit_tuple_type` or `visit_typeddict_type` below, but I wanted to add a runtime check mostly for peace of mind.

Reply:
Alas, it doesn't: I get a 'Argument "final_value" to "Instance" has incompatible type "Optional[Type]"; expected "Optional[LiteralType]"` error a little later on when we use the variable.